plugin/skywalking/plugin-skywalking.go

```diff
@@ -0,0 +1,125 @@
+package skywalking
+
+import (
+	"context"
+	"sync"
+
+	ofctx "github.com/OpenFunction/functions-framework-go/context"
+	"github.com/OpenFunction/functions-framework-go/plugin"
+	"github.com/SkyAPM/go2sky"
+	"github.com/SkyAPM/go2sky/reporter"
+	"k8s.io/klog/v2"
+
+	agentv3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	name    = "skywalking"
+	version = "v1"
+)
+
+var (
+	initGo2skyOnce sync.Once
+)
+
+type klogWrapper struct {
+}
+
+func (k klogWrapper) Info(args ...interface{}) {
+	klog.Info(args)
+}
+
+func (k klogWrapper) Infof(format string, args ...interface{}) {
+	klog.Infof(format, args)
+}
+
+func (k klogWrapper) Warn(args ...interface{}) {
+	klog.Warning(args)
+}
+
+func (k klogWrapper) Warnf(format string, args ...interface{}) {
+	klog.Warningf(format, args)
+}
+
+func (k klogWrapper) Error(args ...interface{}) {
+	klog.Error(args)
+}
+
+func (k klogWrapper) Errorf(format string, args ...interface{}) {
+	klog.Errorf(format, args)
+}
+
+func initGo2sky(ofCtx ofctx.Context, p *PluginSkywalking) {
+	initGo2skyOnce.Do(func() {
+		r, err := reporter.NewGRPCReporter(ofCtx.PluginsTracing.Provider.OapServer, reporter.WithLog(&klogWrapper{}))
+		if err != nil {
+			klog.Errorf("new go2sky grpc reporter error\n", err)
+			return
+		}
+		tracer, err := go2sky.NewTracer(ofCtx.Name, go2sky.WithReporter(r), go2sky.WithInstance(ofCtx.PluginsTracing.Tags["instance"]))
+		if err != nil {
+			klog.Errorf("new go2sky tracer error\n", err)
+			return
+		}
+		go2sky.SetGlobalTracer(tracer)
+
+		p.tracer = tracer
+	})
+}
+
+type PluginSkywalking struct {
+	tracer *go2sky.Tracer
+}
+
+func (p *PluginSkywalking) Init() plugin.Plugin {
+	return p
+}
+
+func (p PluginSkywalking) Name() string {
+	return name
+}
+
+func (p PluginSkywalking) Version() string {
+	return version
+}
+
+func (p *PluginSkywalking) ExecPreHook(ctx ofctx.Context, plugins map[string]plugin.Plugin) error {
+	initGo2sky(ctx, p)
+	if p.tracer == nil {
+		return nil
+	}
+
+	if ctx.SyncRequestMeta.Request != nil {
+		// SyncRequest
+		return preSyncRequestLogic(&ctx, p.tracer)
+	}
+	return nil
+}
+
+func (p *PluginSkywalking) ExecPostHook(ctx ofctx.Context, plugins map[string]plugin.Plugin) error {
```

Comment:
Do you need to replace the Context with *Context in this PR?

Reply:
Yes

Reply:
I think we should submit a new PR to modify and abstract the Context interface

Reply:
Ok

Reply:
Use pointer is better to avoid additional overhead